core/lib/zksync_core/src/proof_data_handler/request_processor.rs

```diff
@@ -140,7 +143,60 @@ impl RequestProcessor {
                     .await
                     .map_err(RequestProcessorError::ObjectStore)?;
 
+                let system_logs_hash_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[0]);
+                let state_diff_hash_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[1]);
+                let bootloader_heap_initial_content_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[2]);
+                let events_queue_state_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[3]);
+
                 let mut storage = self.pool.access_storage().await.unwrap();
+
+                // let header = storage
+                //     .blocks_dal()
+                //     .get_l1_batch_header(l1_batch_number)
+                //     .await
+                //     .unwrap()
+                //     .expect("Proved block without a header");
+
+                let l1_batch = storage
+                    .blocks_dal()
+                    .get_l1_batch_metadata(l1_batch_number)
+                    .await
+                    .unwrap()
+                    .expect("Proved block without metadata");
+
+                let events_queue_state = l1_batch
+                    .metadata
+                    .events_queue_commitment
+                    .expect("No events_queue_commitment");
+                let bootloader_heap_initial_content = l1_batch
+                    .metadata
+                    .bootloader_initial_content_commitment
+                    .expect("No bootloader_initial_content_commitment");
+                let system_logs = serialize_commitments(&l1_batch.header.system_logs);
+                let system_logs_hash = H256(keccak256(&system_logs));
+
+                let state_diff_hash = l1_batch
+                    .header
+                    .system_logs
+                    .into_iter()
+                    .find(|elem| elem.key == u256_to_h256(2.into()))
+                    .expect("No state diff hash key")
+                    .value;
+
+                if events_queue_state != events_queue_state_from_prover
+                    || bootloader_heap_initial_content
+                        != bootloader_heap_initial_content_from_prover
+                    || state_diff_hash != state_diff_hash_from_prover
+                    || system_logs_hash != system_logs_hash_from_prover
+                {
+                    let server_values = format!("{system_logs_hash} {state_diff_hash} {events_queue_state} {bootloader_heap_initial_content}");
+                    let prover_values = format!("{system_logs_hash_from_prover} {state_diff_hash_from_prover} {events_queue_state_from_prover} {bootloader_heap_initial_content_from_prover}");
+                    tracing::error!("Auxilary output doesn't match, server values: {server_values} prover values: {prover_values}");
```

Comment:
so the values don't match, but we still continue processing the proof?

Reply:
+1, panicking might be better we can keep the error log.